plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3RepositoryPlugin.java

```diff
@@ -78,18 +98,62 @@ public class S3RepositoryPlugin extends Plugin implements RepositoryPlugin, Relo
     }
 
     protected final S3Service service;
+    private final S3AsyncService s3AsyncService;
+
     private final Path configPath;
 
+    private AsyncExecutorBuilder priorityExecutorBuilder;
+    private AsyncExecutorBuilder normalExecutorBuilder;
+
     public S3RepositoryPlugin(final Settings settings, final Path configPath) {
-        this(settings, configPath, new S3Service(configPath));
+        this(settings, configPath, new S3Service(configPath), new S3AsyncService(configPath));
+    }
+
+    @Override
+    public List<ExecutorBuilder<?>> getExecutorBuilders(Settings settings) {
+        List<ExecutorBuilder<?>> executorBuilders = new ArrayList<>();
+        executorBuilders.add(new FixedExecutorBuilder(settings, PRIORITY_FUTURE_COMPLETION, 4, 10_000, PRIORITY_FUTURE_COMPLETION));
+        executorBuilders.add(new FixedExecutorBuilder(settings, PRIORITY_STREAM_READER, 4, 10_000, PRIORITY_STREAM_READER));
+        executorBuilders.add(new FixedExecutorBuilder(settings, FUTURE_COMPLETION, 1, 10_000, FUTURE_COMPLETION));
+        executorBuilders.add(new FixedExecutorBuilder(settings, STREAM_READER, 1, 10_000, STREAM_READER));
```

Comment:
why are we using fixed size builder and not `ScalingExecutorBuilder` ?  As we have more number of cores and more indexing subsequently , scaling the threadpool can help with the threadpool